File: JavaNgArch/src/javangarch/bd/AdminBD.java


package javangarch.bd;



import javax.annotation.Resource;



import projectbase.bd.BaseBusinessDelegate;

import projectbase.domain.IGenericDao;
import projectbase.domain.IUtilQuery;
import projectbase.utils.HqlCriterion;
import javangarch.domain.bdinterface.IAdminBD;
import javangarch.domain.domainmodel.gn.Corp;
import javangarch.domain.domainmodel.gn.User;

    public class AdminBD  extends  BaseBusinessDelegate implements IAdminBD
    {
    	
    	@Resource public IGenericDao<User> UserDao;
    	@Resource public IGenericDao<Corp> CorpDao;
    	@Resource  public IUtilQuery UtilQuery;

         public User GetLoginUser(String corpCode,String userCode,String password)
         {
        	 HqlCriterion filter=new HqlCriterion()
        	 .And("this.corp.code=?" , corpCode)
        	 .And("this.code=?" , userCode)
        	 .And("this.password=?",password);
             return UserDao.GetOneByQuery(filter);
         }
         public void RefreshUser(User user)
         {
             UserDao.Refresh(user);
         }
         public void RefreshCorp(Corp corp)
         {
             CorpDao.Refresh(corp);
         }
    }

File: NetNgArch/src/NetNgArch.BusinessDelegate/AdminBD.cs
using System;
using System.Collections.Generic;
using System.Linq;
using NetNgArch.Domain.BDInterface;
using NetNgArch.Domain.DomainModel.GN;
using ProjectBase.BusinessDelegate;
using ProjectBase.Domain;
using ProjectBase.Utils;

namespace NetNgArch.BusinessDelegate
{
    public class AdminBD : BaseBusinessDelegate,IAdminBD
    {
        public IGenericDao<User> UserDao { get; set; }
        public IGenericDao<Corp> CorpDao { get; set; }
        public IUtilQuery UtilQuery { get; set; }

        public User GetLoginUser(string userCode,string password)
        {
            return UserDao.GetOneByQuery(o => o.Code == userCode && o.Password == password);
        }
        public void RefreshUser(User user)
        {
            UserDao.Refresh(user);
        }
        public void RefreshCorp(Corp corp)
        {
            CorpDao.Refresh(corp);
        }
    }
}
